go/worker/client/service.go

```diff
@@ -238,6 +239,35 @@ func (s *service) GetTransactionsWithResults(ctx context.Context, request *api.G
 	return results, nil
 }
 
+// Implements api.RuntimeClient.
+func (s *service) GetUnconfirmedTransactions(_ context.Context, runtimeID common.Namespace) ([][]byte, error) {
+	rt := s.w.commonWorker.GetRuntime(runtimeID)
+	if rt == nil {
+		return nil, api.ErrNotFound
+	}
+
+	// Get current proposed transactions from the runtime's transaction pool.
+	proposedTxs := rt.TxPool.GetProposedTxs()
```

Comment:
Hm, this doesn't seem right, so proposed transactions are just those that have been proposed for the current batch (e.g. on executor nodes). You should instead be getting the pending transactions from all of the different queues, similar to what `GetSchedulingSuggestion` is doing (but without holding the lock).

Reply:
OK, I'll fix it. That part of the code is very confusing.

Reply:
Please re-review :)